mlflow/sklearn/__init__.py

```diff
@@ -455,6 +460,323 @@ def load_model(model_uri):
     )
 
 
+class _AutologTrainingStatus:
+    """
+    This class is designed for holding information which is used by autologging metrics
```

Comment:
Nit: To reduce the file size of `mlflow/sklearn/__init__.py`, could we file a follow-up PR to:

1. Convert `mlflow.sklearn.utils` into a directory with an `__init__.py` file, where we move the contents from `mlflow/sklearn/utils.py` into `mlflow/sklearn/utils/__init__.py`
2. Move the metrics logic introduced by this PR into `mlflow/sklearn/utils/metrics_utils.py`

Reply:
create followup ticket for this https://databricks.atlassian.net/browse/ML-16603

Reply:
Thank you!